playground/frontend/lib/modules/editor/components/pipeline_options_dropdown/pipeline_options_dropdown_body.dart

```diff
@@ -0,0 +1,229 @@
+    if (pipelineOptionsList.isEmpty) {
+      pipelineOptionsList = kDefaultOption;
+    }
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    tabController.removeListener(onTabChange);
+    tabController.dispose();
+    super.dispose();
+  }
+
+  onTabChange() {
+    setState(() {
+      selectedTab = tabController.index;
+    });
+    if (tabController.index == kRawTabIndex) {
+      _updateRawValue();
+    } else {
+      _updateFormValue();
+    }
+  }
+
+  onDelete(int index) {
+    setState(() {
+      pipelineOptionsList.removeAt(index);
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    AppLocalizations appLocale = AppLocalizations.of(context)!;
+    return Column(
+      children: [
+        TabBar(
+          controller: tabController,
+          tabs: <Widget>[
+            Tab(text: appLocale.optionsPipelineOptions),
+            Tab(text: appLocale.rawPipelineOptions),
+          ],
+        ),
+        const PipelineOptionsDropdownSeparator(),
+        Expanded(
+          child: Padding(
+            padding: const EdgeInsets.all(kXlSpacing),
+            child: TabBarView(
+              controller: tabController,
+              physics: const NeverScrollableScrollPhysics(),
+              children: <Widget>[
+                PipelineOptionsForm(
+                  options: pipelineOptionsList,
+                  onDelete: onDelete,
+                ),
+                PipelineOptionsDropdownInput(
+                  controller: pipelineOptionsController,
+                ),
+              ],
+            ),
+          ),
+        ),
+        const PipelineOptionsDropdownSeparator(),
+        Padding(
+          padding: const EdgeInsets.all(kXlSpacing),
+          child: Row(
+            crossAxisAlignment: CrossAxisAlignment.center,
+            children: [
+              SizedBox(
+                height: kButtonHeight,
+                child: ElevatedButton(
+                  child: Text(appLocale.saveAndClose),
+                  onPressed: () => _save(context),
+                ),
+              ),
+              const SizedBox(width: kLgSpacing),
+              if (selectedTab == kOptionsTabIndex)
+                SizedBox(
+                  height: kButtonHeight,
+                  child: OutlinedButton(
+                    child: Text(appLocale.addPipelineOptionParameter),
+                    onPressed: () => setState(() {
+                      pipelineOptionsList.add(PipelineOptionController());
+                    }),
+                  ),
+                ),
+              if (showError && selectedTab == kRawTabIndex)
+                Flexible(
+                  child: Text(
+                    appLocale.pipelineOptionsError,
+                    style: Theme.of(context)
+                        .textTheme
+                        .caption
+                        ?.copyWith(color: kErrorNotificationColor),
```

Comment:
Should this question mark have a default value? E.g. `?? ThemeData.fallback`

Reply:
I updated it to the ```Theme.of(context)
                        .textTheme
                        .caption
                        !.copyWith(color: kErrorNotificationColor)```
because it shouldn't be null here.